backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchBackend.java

```diff
@@ -23,4 +30,22 @@ public interface ElasticsearchBackend extends Backend {
 	 */
 	<T> T client(Class<T> clientClass);
 
+	/**
+	 * Looks up the built-in and configured analyzers available to the index configured for this backed.
+	 * <p>
+	 * This is similar to {@link IndexDescriptor#analyzerDescriptor(String)}.
+	 *
+	 * @param indexName The name of the index.
+	 * @param analyzerName The name of the analyzer.
+	 * @return An {@link Optional#empty() empty optional} if there is no analyzer configured with the given index and name.
+	 */
+	@Incubating
+	Optional<ElasticsearchAnalyzerDescriptor> analyzerDescriptor(String indexName, String analyzerName);
+
+	/**
+	 * @return A map of built-in and configured analyzer descriptors by indexes configured for this backend.
+	 */
+	@Incubating
+	Map<String, Collection<ElasticsearchAnalyzerDescriptor>> analyzerDescriptors();
```

Comment:
This feels dangerous, as there may be conflicts: analyzers with the same name in multiple indexes, but a different definition.

I mean right now it's okay since you only expose the analyzer name, but the possibility of such conflicts will prevent this API from evolving, and makes the whole concept of an "analyzer descriptor" pointless.

IMO you should not expose anything on ElasticsearchBackend. What you added to `IndexDescriptor` is enough, and if one day we need more, we'll make `IndexDescriptor` evolve to expose backend-specific stuff (through extensions or other).

Reply:
> analyzers with the same name in multiple indexes

right... that's why I thought a map <index-name, list-of-analyzers> would work here, but anyways 😃 I'll drop the backend API changes